### PR TITLE
fix(48556): never cannot be spread into object, but it can be spread into JSX props

### LINKED ISSUE #48556: `never` cannot be spread into object, but it can be spread into JSX props
# Bug Report

### 🔎 Search Terms

- props
- never
- spread

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about jsx, `never`, and spreads. This seems like a design limitation.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEAEBcGcFoEsDmA7A9gJwKYCgDGrlpJRUAjAKwC5RlMA3TdUAXlAG8BfbPAokigEwt2AOjFlyXfIWKYANsIA8AE3h1R4ih1DAAfEA)

### 💻 Code

```ts
// @ts-ignore
const obj: never = {}

const obj2 = {...obj}
const el = <div {...obj} />
```

### 🙁 Actual behavior

As seen in https://github.com/kiliman/remix-params-helper/pull/19, `never` is able to be spread into JSX props, even though `never` cannot be spread into objects. This seems odd to me as both compile to `Object.assign`. See screenshots:

TypeScript prevents spreading `never` into an object:

image

But spreading `never` into props works fine:

image

### 🙂 Expected behavior

`never` should be prevented from spreading into jsx props, just like it cannot be spread into objects.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #48556

### CODE REVIEW

**Review by sandersn — APPROVED** [review]
I think this is right, but while I'm looking at it, it's kind of strange that object literals return `errorType` whereas JSX literals try to build up `typeToIntersect` as a place to hold non-spreadable types. I would think that both kinds of literals would behave the same.

**Comment by DanielRosenwasser** [comment]
This breaks spreading `unknown` and `never` into a JSX element. Users may now see

> Spread types may only be created from object types.

In JSX spreads.

File: tests/baselines/reference/jsxExcessPropsAndAssignability.errors.txt [hunk 1]
```diff
@@ -17,10 +19,14 @@ tests/cases/compiler/jsxExcessPropsAndAssignability.tsx(16,6): error TS2322: Typ
     
         // Expected no error, got none - good
```
Comment (Jessidhia) [inline tests/baselines/reference/jsxExcessPropsAndAssignability.errors.txt:20]: This comment now seems inaccurate 😅

**Comment by nandorojo** [comment]
Can you still spread a nullable object onto JSX?

**Comment by DanielRosenwasser** [comment]
I believe so

**Pull request merged**